src/windows/BarcodeScannerProxy.js

```diff
@@ -432,21 +432,39 @@ module.exports = {
          * See https://github.com/phonegap-build/BarcodeScanner#using-the-plugin
          */
         function cancelPreview() {
+            BarcodeReader.scanCancelled = true;
             reader && reader.stop();
         }
 
+        function checkCancelled() {
+            if (BarcodeReader.scanCancelled) {
+                throw new Error('Canceled');
+            }
+        }
+
+        if (BarcodeReader.scanCancelled) {
+            success({
+                cancelled: true
+            });
+            BarcodeReader.scanCancelled = false;
```

Comment:
Probably it's more convenient to set this to `false` in `createPreview` once - just before binding `cancelPreview` to back and Cancel buttons